lib/rester/client/adapters/local_adapter.rb

```diff
@@ -37,11 +37,13 @@ def _request(verb, path, opts={})
 
         response = Timeout::timeout(timeout) do
           service.call(
-            'REQUEST_METHOD' => verb.to_s.upcase,
-            'PATH_INFO'      => path,
-            'CONTENT_TYPE'   => 'application/x-www-form-urlencoded',
-            'QUERY_STRING'   => query,
-            'rack.input'     => StringIO.new(body)
+            {
+              'REQUEST_METHOD' => verb.to_s.upcase,
+              'PATH_INFO'      => path,
+              'CONTENT_TYPE'   => 'application/x-www-form-urlencoded',
+              'QUERY_STRING'   => query,
+              'rack.input'     => StringIO.new(body)
+              }.merge(opts[:headers])
```

Comment:
Should format headers in the `HTTP_` format.  Normally headers won't be in that format, so this should do it behind the scenes.

Also, merge into the headers, instead of the other way around. That way the headers can't override any of the other values:
```ruby
opts[:headers].merge(
  'REQUEST_METHOD' => verb.to_s.upcase,
  'PATH_INFO'      => path,
  'CONTENT_TYPE'   => 'application/x-www-form-urlencoded',
  'QUERY_STRING'   => query,
  'rack.input'     => StringIO.new(body)
)
```